packages/create-astro/src/actions/context.ts

```diff
@@ -34,6 +34,13 @@ export interface Context {
 }
 
 export async function getContext(argv: string[]): Promise<Context> {
+	const packageSpecifier = argv
+		.find((argItem) => /^(astro|create-astro)@/.exec(argItem))
+		?.split('@')[1];
+	// Fallback to 'latest' if it's a version number
+	const packageTag =
+		packageSpecifier && /^v?\d[^a-zA-Z]*$/.test(packageSpecifier) ? 'latest' : packageSpecifier;
```

Comment:
Should we play a bit safer here and only allow known tags? e.g.

```js
switch (packageSpecifier) {
  case 'alpha':
  case 'beta':
  case 'rc':
    return packageManager;
  default:
    return 'latest'
}
```

The code here is inferring the tag between `create-astro` and `astro`, and they might not match if we're loosely passing them.